File: TroopersApp/app/src/main/java/com/example/teffe/troopersapp/util/ResourceUtil.java
package com.example.teffe.troopersapp.util;

import com.example.teffe.troopersapp.R;
import com.example.teffe.troopersapp.model.Affiliation;

/**
 * Created by teffe on 18/11/2017.
 */

public class ResourceUtil {
    public static int getResourceBasedOnAffiliation(Affiliation affiliation) {
        switch (affiliation) {
            case GALACTIC_REPUBLIC:
                return R.drawable.galactic_republic;
            case GALACTIC_EMPIRE:
                return R.drawable.galactic_empire;
            case FIRST_ORDER:
                return R.drawable.first_order;
            default:
                return 0;
        }
    }
}

File: TroopersApp/app/src/main/java/com/example/teffe/troopersapp/util/ResourceUtil.kt
package com.example.teffe.troopersapp.util

import com.example.teffe.troopersapp.R
import com.example.teffe.troopersapp.model.Affiliation

/**
 * Created by teffe on 20/11/2017.
 */
object ResourceUtil {
    fun getResourceBasedOnAffiliation(affiliation: Affiliation?): Int {
        return when (affiliation) {
            Affiliation.GALACTIC_EMPIRE -> R.drawable.galactic_empire
            Affiliation.GALACTIC_REPUBLIC -> R.drawable.galactic_republic
            Affiliation.FIRST_ORDER -> R.drawable.first_order
            else -> 0
        }
    }

    fun getNameBasedOnAffiliation(affiliation: Affiliation?): Int {
        return when (affiliation) {
            Affiliation.FIRST_ORDER -> R.string.first_order
            Affiliation.GALACTIC_EMPIRE -> R.string.galactic_empire
            Affiliation.GALACTIC_REPUBLIC -> R.string.galactic_republic
            else -> 0
        }
    }
}